src/main/java/com/combatlogger/CombatLoggerPlugin.java

```diff
@@ -650,71 +650,74 @@ public void onMenuOpened(MenuOpened event) {
 		Point mousePosition = client.getMouseCanvasPosition();
 		Rectangle overlayBounds = damageOverlay.getBounds();
 
-		if (overlayBounds.contains(mousePosition.getX(), mousePosition.getY())) {
-			// Get existing menu entries
-			MenuEntry[] existingEntries = event.getMenuEntries();
+		if(!overlayVisible || !overlayBounds.contains(mousePosition.getX(), mousePosition.getY())){
+			return;
+		}
```

Comment:
This is the only section that changed, the rest is formatting fluff.

Prevents right click contextmenu options from being added if the overlay isn't visible.